### PR TITLE
Support Windows build target

### PR DESCRIPTION
**Pull request by wohali**

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -9,5 +9,7 @@
     % {".*", "CFLAGS", "$CFLAGS -g -Wall -Werror -fPIC"}
 
     % Production compilation
-    {".*", "CFLAGS", "$CFLAGS -Wall -Werror -DNDEBUG -O3"}
+    {"(linux|solaris|darwin|freebsd)", "CFLAGS", "$CFLAGS -Wall -Werror -DNDEBUG -O3"},
+    {"win32", "CFLAGS", "$CFLAGS /O2 /DNDEBUG /Wall"}
```
Comment (kxepal) [inline rebar.config:13]: Does it matter that -03 applied for everything, but for Windows is only -02?
Also, may be put /we key for windows to have warnings-as-errors enabled?

Comment (wohali) [inline rebar.config:13]: Windows doesn't have a -O3 flag for MSVC, only -O2.

You don't want warnings-as-errors, the code will never successfully compile. We do a lot of naughty things and I don't want cleanup of that code on all platforms to block our release.

Comment (kxepal) [inline rebar.config:13]: Ok.

**Comment by kxepal** [comment]
+1

**Comment by kocolosk** [comment]
+1

**Comment by NorthNick** [comment]
+1

**Pull request closed without merging**